url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/com/mxgraph/util/mxCurve.java

```diff
@@ -0,0 +1,1244 @@
+
+		// If any of the new points are null, ignore the list
+		for (mxPoint point : newPoints)
+		{
+			if (point == null)
+			{
+				return;
+			}
+		}
+
+		if (newPoints.size() != guidePoints.size())
+		{
+			pointsChanged = true;
+		}
+		else
+		{
+			// Check for a constant translation of all guide points. In that 
+			// case apply the translation directly to all curves.
+			// Also check whether all of the translations are trivial
+			if (newPoints.size() == guidePoints.size()
+					&& newPoints.size() > 1 && guidePoints.size() > 1)
+			{
+				boolean constantTranslation = true;
+				boolean trivialTranslation = true;
+				mxPoint newPoint0 = newPoints.get(0);
+				mxPoint oldPoint0 = guidePoints.get(0);
+				double transX = newPoint0.getX() - oldPoint0.getX();
+				double transY = newPoint0.getY() - oldPoint0.getY();
+
+				if (Math.abs(transX) > 0.01 || Math.abs(transY) > 0.01)
+				{
+					trivialTranslation = false;
+				}
+
+				for (int i = 1; i < newPoints.size(); i++)
+				{
+					double nextTransX = newPoints.get(i).getX()
+							- guidePoints.get(i).getX();
+					double nextTransY = newPoints.get(i).getY()
+							- guidePoints.get(i).getY();
+
+					if (Math.abs(transX - nextTransX) > 0.01 || Math.abs(transY - nextTransY) > 0.01)
+					{
+						constantTranslation = false;
+					}
+
+					if (Math.abs(nextTransX) > 0.01
+							|| Math.abs(nextTransY) > 0.01)
+					{
+						trivialTranslation = false;
+					}
+				}
+
+				if (trivialTranslation)
+				{
+					pointsChanged = false;
+				}
+				else if (constantTranslation)
+				{
+					pointsChanged = false;
+					// Translate all stored points by the translation amounts
+					Collection<mxPoint[]> curves = points.values();
+
+					// Update all geometry information held by the curve
+					// That is, all the curve points, the guide points
+					// and the cached bounds
+					for (mxPoint[] curve : curves)
+					{
+						for (int i = 0; i < curve.length; i++)
+						{
+							curve[i].setX(curve[i].getX() + transX);
+							curve[i].setY(curve[i].getY() + transY);
+						}
+					}
+
+					guidePoints = new ArrayList<mxPoint>(newPoints);
+					minXBounds += transX;
+					minYBounds += transY;
+					maxXBounds += transX;
+					maxYBounds += transY;
+				}
+				else
+				{
+					pointsChanged = true;
+				}
+			}
+		}
+
+		if (pointsChanged)
+		{
+			guidePoints = new ArrayList<mxPoint>(newPoints);
+			points = new Hashtable<String, mxPoint[]>();
+			valid = false;
+		}
+	}
+
+	/**
+	 * Obtains the points that make up the curve for the specified
+	 * curve index. If that curve, or the core curve that other curves
+	 * are based on have not yet been created, then they are lazily
+	 * created. If creation is impossible, null is returned
+	 * @param index the key specifying the curve
+	 * @return the points making up that curve, or null
+	 */
+	public mxPoint[] getCurvePoints(String index)
+	{
+		if (validateCurve())
+		{
+			if (points.get(LABEL_CURVE) == null && index == LABEL_CURVE)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.